Dockerfile

```diff
@@ -0,0 +1,73 @@
+# build ironlib wrapper binaries
+FROM golang:1.18-alpine AS stage0
+
+WORKDIR /workspace
+
+# copy the go modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# copy ironlib go sources
+COPY . .
+
+# build helper util
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on \
+    go build -o getbiosconfig examples/biosconfig/biosconfig.go && \
+    install -m 755 -D getbiosconfig /usr/sbin/
+
+FROM almalinux:9-minimal
+
+ARG DEPDIR="dependencies"
+COPY "${DEPDIR}" dependencies
+
+# copy ironlib wrapper binaries
+COPY --from=stage0 /usr/sbin/getbiosconfig /usr/sbin/getbiosconfig
+
+# import and install tools
+RUN microdnf install -y --setopt=tsflags=nodocs crypto-policies-scripts && \
+    update-crypto-policies --set DEFAULT:SHA1 && \
+    rpm -ivh ${DEPDIR}/epel-release-latest-9.noarch.rpm
+
+## Prerequisite directories for Dell, ASRR, Supermicro
+## /lib/firmware required for Dell updates to be installed successfullly
+RUN mkdir -p /lib/firmware /opt/asrr /usr/libexec/dell_dup /opt/supermicro/sum/
+# Bootstrap Dell DSU repository
+RUN curl -O https://linux.dell.com/repo/hardware/dsu/bootstrap.cgi && bash bootstrap.cgi && rm -f bootstrap.cgi
+# Install Dell idracadm7 to enable collecting BIOS configuration and use install_weak_deps=0 to avoid pulling extra packages
+RUN microdnf install -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
+    srvadmin-idracadm7
+
+# install misc support packages
+RUN microdnf install -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 \
+    lshw          \
+    pciutils      \
+    hdparm        \
+    util-linux    \
+    nvme-cli      \
+    dmidecode     \
+    libssh2-devel \
+    kmod          \
+    tar           \
+    gzip          \
+    unzip         \
+    which         \
```

Comment:
I find it very nice to indent the long-list-of-things for a command by a couple of spaces so that the indentation helps with grouping.
So something like:
```
RUN microdnf ... \
      lshw \
      ...
      smartmontools && \
    gdisk && \
    ...
```
Also sorting long lists is nice too :D

Reply:
Oh another thing that I've found to be nice when having a long list in the middle of && \ chains in Dockerfiles is to end the list with a solo `;` this way you can add new things at the end and its only one line of diff:

```
RUN microdnf ... \
      lshw \
      ...
      smartmontools \
      ; && \
    gdisk && \
```

Reply:
I'll get that fixed before the final PR is ready. I got the list broken when adding the latest changes done in original Dockerfile

Reply:
Fixed in [a3b0d66](https://github.com/metal-toolbox/ironlib/pull/84/commits/a3b0d668acd8efe2ed5c1d465685196963d0b747)